library/std/src/f32/tests.rs

```diff
@@ -287,6 +287,69 @@ fn test_is_sign_negative() {
     assert!((-f32::NAN).is_sign_negative());
 }
 
+#[test]
```

Comment:
Maybe this ? This seems enough to [only exclude](https://godbolt.org/z/KarxqzEna) `i586-unknown-linux-gnu`.
```suggestion
#[test]
#[cfg_attr(not(all(target_arch = "x86", not(target_feature = "fxsr"))), ignore)]
```

Reply:
What is the `fxsr` target feature? Either way I would only exclude the NaN block from the test, not the whole test.

Reply:
From what I can tell this feature tell if these float `FXSAVE, FXRESTOR` instructions are available.
This CPU feature is not be available for the `pentium` cpu which is the base cpu of `i586-unknown-linux-gnu` but is available after that. This was the only difference I could find to only exclude the `i586-unknown-linux-gnu` target.

Reply:
I don't want to skip the entire test case on this platform, only the NaN round-trip, so I'm just going to conditionally compile out the NaN payload block only.